CMakeLists.txt

```diff
@@ -1,86 +1,66 @@
-cmake_minimum_required(VERSION 3.15)
-
-# Set the project name to your project name, my project isn't very descriptive
-project(myproject CXX)
-include(cmake/StandardProjectSettings.cmake)
-include(cmake/PreventInSourceBuilds.cmake)
-
-# Link this 'library' to set the c++ standard / compile-time options requested
-add_library(project_options INTERFACE)
+cmake_minimum_required(VERSION 3.16)
+
+# uncomment to set a default CXX standard for the external tools like clang-tidy and cppcheck
+# and the targets that do not specify a standard.
+# If not set, the latest supported standard for your compiler is used
+# You can later set fine-grained standards for each target using `target_compile_features`
+# set(CMAKE_CXX_STANDARD 17)
+
+# Add project_options v0.12.1
+# https://github.com/cpp-best-practices/project_options
+include(FetchContent)
+FetchContent_Declare(_project_options URL https://github.com/cpp-best-practices/project_options/archive/refs/tags/v0.12.1.zip)
+FetchContent_MakeAvailable(_project_options)
+include(${_project_options_SOURCE_DIR}/Index.cmake)
+
+# uncomment to enable vcpkg:
+# # Setup vcpkg - should be called before defining project()
+# run_vcpkg()
+
+# Set the project name and language
+project(myproject LANGUAGES CXX)
+
+# Initialize project_options
+# uncomment the options to enable them:
+project_options(
```

Comment:
Apologies in advance if this critique sounds nit-picky, because it is definitely nit-picky.

I think we have a name collision here. As I understand it, we have two variables named `project_options`:
* the macro called `project_options`, that we imported from the `project_options` library
* the 'library interface' called `project_options`, that was created when we called the `project_options` macro

To my understanding, the library interface `project_options` object will replace the macro `project_options`, and every time you reference `project_options` after this point, you will get the library interface and not the macro. This should work just fine, and it will not cause any issue unless you decide you want to use the `project_options` macro again.

I think that this could be very difficult for a first-time user of the project to understand. To understand this code, you are forced to dig into the 'project_options' source code. I think it would be easier to understand if the names of the macro and the library interface were different, to make it clear that they are not the same thing.

I think the `project_options` macro could be easier to understand if it allowed the user to set the name of the library interface, like this:

```cmake
# create `my_project_options_interface_target`
project_options(                         # call the macro
  my_project_options_interface_target    # name of the library interface to create
  ENABLE_CACHE                           # option to turn on
)

# use `my_project_options_interface_target`
target_compile_features(my_project_options_interface_target INTERFACE cxx_std_17)
add_executable(main main.cpp)
target_link_libraries(main PRIVATE my_project_options_interface_target)
```

Reply:
Thanks, I thought about it. But I am not sure if we want allow calling project_options twice the same way Cmake allows only one project.

Reply:
Maybe we could update the comment to explain what's going on, so that new users don't get the idea that this call to the `project_options()` macro is a constructor?

Also, the comment below this still refers to `cmakelib`.

Reply:
Fixed in https://github.com/cpp-best-practices/cpp_starter_project/pull/169/commits/4221006d809b899174588a7581e8ada2f0a53c4e